### PR TITLE
chore(bazel): enable rules_esbuild sandbox with object-inspect workaround

### PR DESCRIPTION
**Pull request by Strum355**
Sandbox escapes be-gone

## Test plan

Tested in CI and locally with `bazel build //client/...` as well as a lot of blood, sweat n tears tearing through failed sandboxes

## Changelog

### CODE REVIEW

File: client/testing/BUILD.bazel [hunk 1]
```diff
@@ -60,7 +60,7 @@ npm_package(
 js_library(
     name = "fetch-mock",
     srcs = ["src/fetch.js"],
-    declarations = ["src/fetch.js"],
+    types = ["src/fetch.js"],
```
Comment (Strum355) [inline client/testing/BUILD.bazel:63]: https://sourcegraph.com/github.com/aspect-build/rules_js/-/blob/js/defs.bzl?L43-50

File: package.json [hunk 2]
```diff
@@ -451,6 +451,21 @@
           "ws": "*"
         }
       },
+      "mz": {
+        "dependencies": {
+          "graceful-fs": "*"
+        }
+      },
```
Comment (Strum355) [inline package.json:458]: Point the finger of shame at this shameful package

File: package.json [hunk 3]
```diff
@@ -451,6 +451,21 @@
...
+      "follow-redirects": {
+        "dependencies": {
+          "debug": "*"
+        }
+      },
+      "debug": {
+        "dependencies": {
+          "supports-color": "*"
+        }
+      },
```
Comment (Strum355) [inline package.json:468]: These are less shameful (the deps are marked as optional in `peerDependenciesMeta` and idk if theres a better way to surface them without adding them directly to our root dependencies list)

File: package.json (hunk 2 above)
Comment (burmudar) [reply package.json:458]: https://www.npmjs.com/package/mz "modernize" ... _grabs pitchfork_ TO ARMS

**Review by burmudar — APPROVED** [review]
For the uninitiated - how did you find the offending packages that did not declare their deps properly?

**Comment by Strum355** [comment]
> For the uninitiated - how did you find the offending packages that did not declare their deps properly?

See the build errors here: https://buildkite.com/sourcegraph/sourcegraph/builds/276958. For better behaved packages, its a case of it being an optional peer dependency (aka one that would be dragged in by another package afaik, so not brought in my default)

File: third_party/rules_esbuild/sandbox-plugin-fixes.patch [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+index c58b668..0ba6cff 100644
+--- a/esbuild/private/plugins/bazel-sandbox.js
++++ b/esbuild/private/plugins/bazel-sandbox.js
+@@ -24,7 +24,15 @@ function bazelSandboxPlugin() {
+           }
+           otherOptions.pluginData.executedSandboxPlugin = true
+
+-          return await resolveInExecroot(build, importPath, otherOptions)
++          const res = await resolveInExecroot(build, importPath, otherOptions)
++          // Needed due to an issue with esbuild when it comes to plguins + `"browser": false` in package.json
```
[the remark this answers is not in the document]
Comment (Strum355) [reply third_party/rules_esbuild/sandbox-plugin-fixes.patch:11]: darn, I completely skimmed this :laughing: didnt notice the typo

**Pull request merged**